node/src/main/scala/coop/rchain/node/web/Transaction.scala

```diff
@@ -69,16 +69,26 @@ final case class TransactionAPIImpl[F[_]: Concurrent](
   ) = findTransactions(report).map(t => TransactionInfo(t, txCtor(d.deployInfo.get.sig)))
 
   def getTransaction(blockHash: Blake2b256Hash): F[List[TransactionInfo]] =
-    blockReportAPI.blockReport(blockHash.toByteString, forceReplay = false).map { res =>
+    blockReportAPI.blockReport(blockHash.toByteString, forceReplay = false).flatMap { res =>
       res
-        .map { b =>
-          val userTransactions = b.deploys.flatMap(d => {
-            d.report
-              .zip(Seq(PreCharge, UserDeploy, Refund))
-              .flatMap {
+        .traverse { b =>
+          val userTransactions = b.deploys.toList.flatTraverse(d => {
```

Comment:
above you use `traverse { b => ` but here `flatTraverse(d => {`
I'd stick to `{ b =>` notation everywhere :)

Reply:
I am ok with that. But it would generate one more line by the formatter.